### PR TITLE
win: implement uv_try_write() for pipes

### PR DESCRIPTION
**Pull request by bnoordhuis**
I can't promise this actually works.  You might want to hold off on reviewing until the CI is green. =)

CI: https://ci.nodejs.org/job/libuv-test-commit/55/

### CODE REVIEW

**Comment by saghul** [comment]
I think this approach cannot be used when the pipe is non-overlapped (ie, it has the UV_HANDLE_NON_OVERLAPPED_PIPE flag set). With those pipes we do the reads / writes in a threadpool.

@piscisaureus knows better, but IIRC this happens when we do stdio over pipes, like the ones on the child process spawned with `uv_spawn`.

File: src/win/pipe.c [hunk 1]
```diff
@@ -1492,6 +1492,68 @@ int uv_pipe_write2(uv_loop_t* loop,
...
+      bytes_written += buf->len;
+      continue;
+    }
+
+    err = GetLastError();
+    if (err != ERROR_IO_PENDING) {
+      break;
+    }
+
+    err = WaitForSingleObject(overlapped.hEvent, 0);
```
Comment (saghul) [inline src/win/pipe.c]: if the pipe has the blocking writes should we wait infinitely?

Comment (bnoordhuis) [inline src/win/pipe.c]: That seems, on the whole, good for consistency with UNIX. I'll add that.

**Comment by bnoordhuis** [comment]
Incorporated feedback in a separate commit for reviewer convenience, PTAL.

CI: https://ci.nodejs.org/job/libuv-test-commit/151/

**Comment by saghul** [comment]
Ping? What's the state of this, Ben?

**Comment by txdv** [comment]
Can you initiate a new test run?

**Comment by bnoordhuis** [comment]
I haven't revisited this and I have no real reason to believe it's going to fare better than last time but rebase + new CI: https://ci.nodejs.org/job/libuv-test-commit/177/

**Comment by txdv** [comment]
***[alpine34-x64](https://ci.nodejs.org/job/libuv-test-commit-linux/203/nodes=alpine34-x64/consoleText)***

* tty_pty
* udp_multicast_join6
* pipe_getsockname_abstract
* poll_bad_fdtype
* socket_buffer_size

***[centos6-64](https://ci.nodejs.org/job/libuv-test-commit-linux/203/nodes=centos6-64/consoleText)***

* getaddrinfo_fail_sync

***[win10-vs2015](https://ci.nodejs.org/job/libuv-test-commit-windows/237/nodes=win10-vs2015/consoleText)***

* pipe_try_write_0
* pipe_try_write_1
* pipe_try_write_2

***[win2008r2-vs2013](https://ci.nodejs.org/job/libuv-test-commit-windows/237/nodes=win2008r2-vs2013/consoleText)***

* pipe_try_write_0
* pipe_try_write_1
* pipe_try_write_2
* threadpool_cancel_single

***[win2012r2-vcbt2015](https://ci.nodejs.org/job/libuv-test-commit-windows/237/nodes=win2012r2-vcbt2015/consoleText)***

* pipe_try_write_0
* pipe_try_write_1
* pipe_try_write_2

**Pull request closed without merging**